pkg/autoscaler/metrics/http_scrape_client.go

```diff
@@ -32,6 +32,54 @@ type httpScrapeClient struct {
 	httpClient *http.Client
 }
 
+var step = 64
+
+var dataSizeClasses = []int{
```

Comment:
Divide pool in classes so that in case we have large and small `podname` strings we dont allocate more than we need for small ones.
This is assumes a randomized size for the podname, maybe in practice smaller pod names appear due to the relationship with the revision name, but this is also user driven. So better be generic and not guess.

Reply:
I'm interested in how the benchmark mem usage changes if we only use the largest class? How much does the multiple pools buy us vs. one fixed-size 304-byte pool?

Reply:
I started with that but I saw better numbers with multiple pools, imagine you get data of size 100 and you constantly allocating 304 maybe, of course its a pool but still. Anyway I can do the experiment.

Reply:
If we decide to do this, we should probably just use named enum constants.